src/ring_theory/dedekind_domain.lean

```diff
@@ -249,4 +255,107 @@ begin
                  fractional_ideal.map_mul, fractional_ideal.map_div, inv_eq] },
 end
 
+lemma is_dedekind_domain_inv.mul_inv_eq_one [algebra A K] [is_fraction_ring A K]
+  (h : is_dedekind_domain_inv A)
+  {I : fractional_ideal A⁰ K} (hI : I ≠ 0) : I * I⁻¹ = 1 :=
+is_dedekind_domain_inv_iff.mp h I hI
+
+lemma is_dedekind_domain_inv.inv_mul_eq_one [algebra A K] [is_fraction_ring A K]
+  (h : is_dedekind_domain_inv A)
+  {I : fractional_ideal A⁰ K} (hI : I ≠ 0) : I⁻¹ * I = 1 :=
+(mul_comm _ _).trans (h.mul_inv_eq_one hI)
+
+lemma is_dedekind_domain_inv.is_unit [algebra A K] [is_fraction_ring A K]
+  (h : is_dedekind_domain_inv A)
+  {I : fractional_ideal A⁰ K} (hI : I ≠ 0) : is_unit I :=
+is_unit_of_mul_eq_one _ _ (h.mul_inv_eq_one hI)
+
+lemma is_dedekind_domain_inv.is_noetherian_ring
+  (h2 : is_dedekind_domain_inv A) : is_noetherian_ring A :=
+begin
+  refine is_noetherian_ring_iff.mpr ⟨λ (I : ideal A), _⟩,
+  by_cases h : I = ⊥,
+  { rw h, apply submodule.fg_bot },
+
```

Comment:
space intentionally left blank?

Reply:
I like whitespaces, but this is indeed a bit too much whitespace.